src/rng.rs

```diff
@@ -1,32 +1,61 @@
 use crate::Vec;
-use algebra::{FromBytes, ToBytes, Field};
+use algebra::{FromBytes, ToBytes, to_bytes, Field};
 use std::{
     convert::TryInto,
-    marker::PhantomData
+    marker::PhantomData,
 };
 use digest::{generic_array::GenericArray, Digest};
 use rand_chacha::ChaChaRng;
 use rand_core::{RngCore, SeedableRng};
 use rand::Rng;
+use crate::error::Error;
 
-/// General trait for `SeedableRng` that refreshes its seed by hashing together the previous seed
-/// and the new seed material.
-// TODO: later: re-evaluate decision about ChaChaRng
-pub trait FiatShamirRng: RngCore {
+/// General trait for an updatable seed to be used to initialize a `FiatShamirRng`.
```

Comment:
Lets comment here 
"A trait for circuit-friendly serialization of Fiat Shamir seed material"